bugzilla/commentsdisk.go

```diff
@@ -160,6 +160,24 @@ func (s *CommentDiskStore) Sync(keys []string) ([]*BugComments, error) {
 	return bugs, nil
 }
 
+func (s *CommentDiskStore) DeleteBug(bug *Bug) error {
+	_, path := s.pathForBug(bug)
+	return os.Remove(path)
+}
+
+func (s *CommentDiskStore) CloseBug(bug *Bug) error {
+
+	_, path := s.pathForBug(bug)
+	bugComments, err := readBugComments(path)
+	if err != nil {
+		return fmt.Errorf("could not read bug at path %s due to error: %v", path, err)
+	}
+	bugComments.Info.Status = "CLOSED"
+	if err := s.write(&Bug{Info: bugComments.Info}, bugComments); err != nil {
```

Comment:
You lost object meta here.  I'm confused why this is necessary?  Why aren't you shallow copying the input bug?  Is this because of tombstones?

Reply:
> Why aren't you shallow copying the input bug? Is this because of tombstones?

if this is what you're calling a tombstone:
```
func (s *CommentStore) bugDelete(obj interface{}) {
	var name string
	var id int
	var err error
	switch t := obj.(type) {
	case cache.DeletedFinalStateUnknown:
		name = t.Key
		id, err = strconv.Atoi(t.Key)
		if err != nil {
			klog.Errorf("Unable to convert string %q to int: %v", t.Key, err)
			return
		}
```

then yes.


And write() doesn't do anything w/ object meta anyway (hence why it's not available to read back either)

I will look into reversing the calls in commentstore.BugDelete so i can fetch the bug from the in memory store before it gets deleted there, but for this function to be truly "safe-ish" it probably still needs to fallback to reading the bug from disk in the event it can't find it in the in memory store.

Reply:
@smarterclayton comments addressed.  There is some risk here that the disk+memory store get out of sync in which case we will not be able to close the bug on disk because we have no record of it in the memory store.  I'm relying on you to tell me if that's worth worrying about or not.